### PR TITLE
chore: updates Accordion to follow design spec tokens

### PR DESCRIPTION
**Pull request by bsunderhus**
## Current Behavior

Some discrepancies were found between Design spec and Accordion implementation,

1. spacing tokens not being used
2. background color token wrong
3. line-height and font-size tokens wrong

## New Behavior

Resolves those discrepancies

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5a92ebd3f8687a1cd40e8233cacdb91d324b742a'>5a92ebd3f8687a1cd40e8233cacdb91d324b742a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/Avatar-mount_base.html">940</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/Avatar-mount.html">931</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/Button-mount_base.html">594</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/Button-mount.html">594</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/FluentProvider-mount_base.html">1994</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/FluentProvider-mount.html">2126</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/FluentProviderWithTheme-mount_base.html">305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/FluentProviderWithTheme-mount.html">291</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">221</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">249</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">320</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">281</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/MakeStyles-mount_base.html">1696</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23295/perf-test-react-components/MakeStyles-mount.html">1756</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by ling1726 — APPROVED** [review]

File: packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeaderStyles.ts [hunk 1]
```diff
@@ -42,51 +42,62 @@ const useStyles = makeStyles({
...
-    paddingRight: '10px',
-    paddingLeft: '10px',
+    ...shorthands.padding(0, tokens.spacingHorizontalM, 0, tokens.spacingHorizontalMNudge),
     height: '44px',
     display: 'flex',
     alignItems: 'center',
     cursor: 'pointer',
-    fontSize: tokens.fontSizeBase300,
-    fontFamily: tokens.fontFamilyBase,
+    ...typographyStyles.body1,
```
Comment (layershifter) [inline packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeaderStyles.ts:52]: Do we need this at all? Are not all of them are defined by `FluentProvider`?

**Review by spmonahan — APPROVED** [review]

**Comment by bsunderhus** [comment]
@microsoft/cxe-coastal  can you please review this?

File: packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeaderStyles.ts [hunk 2]
```diff
@@ -42,51 +42,62 @@ const useStyles = makeStyles({
   },
   button: {
     position: 'relative',
-    width: 'calc(100% - 22px)',
+    width: '100%',
     ...shorthands.border('1px', 'solid', 'transparent'),
-    paddingRight: '10px',
-    paddingLeft: '10px',
+    ...shorthands.padding(0, tokens.spacingHorizontalM, 0, tokens.spacingHorizontalMNudge),
```
[the remark this answers is not in the document]
Comment (bsunderhus) [reply packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeaderStyles.ts:47]: Yeah, I can confirm 😅, make no sense, right? hehehehe

**Pull request merged**